### PR TITLE
[FLINK-7907][docs] The metrics documentation missing scala snippets

### PR DESCRIPTION
**Pull request by yew1eb**
## What is the purpose of the change

*The metrics documentation is missing Scala examples for many of the metrics types. To be consistent there should be Scala examples for all the types.*

## Brief change log

  - *Add scala snippets for  `Counter` , `Histogram `, `Meter` types of metrics*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: docs/monitoring/metrics.md [hunk 1]
```diff
@@ -61,9 +63,35 @@ public class MyMapper extends RichMapFunction<String, String> {
 }
 
 {% endhighlight %}
+</div>
+
+<div data-lang="scala" markdown="1">
+{% highlight scala %}
+
+public class MyMapper extends RichMapFunction[String,String] {
```
Comment (zentol) [inline docs/monitoring/metrics.md]: remove public keyword (also applies to remaining examples)

File: docs/monitoring/metrics.md [hunk 2]
```diff
@@ -61,9 +63,35 @@ public class MyMapper extends RichMapFunction<String, String> {
...
+  var counter: Counter
```
Comment (zentol) [inline docs/monitoring/metrics.md]: reasonable sure this must be lazy (also applies to remaining examples)

**Comment by yew1eb** [comment]
@zentol , thanks for your review. I have updated the PR according to your comments.

File: docs/monitoring/metrics.md [hunk 3]
```diff
@@ -94,8 +94,8 @@ Alternatively you can also use your own `Counter` implementation:
 <div data-lang="java" markdown="1">
 {% highlight java %}
 
-public class MyMapper extends RichMapFunction<String, String> {
-  private Counter counter;
+RichMapFunction<String, String> {
```
Comment (zentol) [inline docs/monitoring/metrics.md]: looks like you removed more than intended?

Comment (yew1eb) [reply docs/monitoring/metrics.md]: Now this PR has been updated, please look at it again.

**Comment by zentol** [comment]
merging.

**Comment by zentol** [comment]
merged, could you close the PR? Thanks!

**Pull request closed without merging**